Sources/Sentry/SentryExtraContextProvider.m

```diff
@@ -0,0 +1,77 @@
+#import "SentryExtraContextProvider.h"
+#import "SentryCrashIntegration.h"
+#import "SentryCrashWrapper.h"
+#import "SentryNSProcessInfoWrapper.h"
+#import "SentryUIDeviceWrapper.h"
+
+@interface
+SentryExtraContextProvider ()
+
+@property (nonatomic, strong) SentryCrashWrapper *crashWrapper;
+@property (nonatomic, strong) SentryUIDeviceWrapper *deviceWrapper;
+@property (nonatomic, strong) SentryNSProcessInfoWrapper *processInfoWrapper;
+
+@end
+
+@implementation SentryExtraContextProvider
+
++ (instancetype)sharedInstance
+{
+    static SentryExtraContextProvider *instance = nil;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{ instance = [[self alloc] init]; });
+    return instance;
+}
+
+- (_Nonnull instancetype)init
+{
+    if (self = [super init]) {
+        self.crashWrapper = [SentryCrashWrapper sharedInstance];
+        self.deviceWrapper = [[SentryUIDeviceWrapper alloc] init];
+        self.processInfoWrapper = [[SentryNSProcessInfoWrapper alloc] init];
```

Comment:
m: I think this 3 properties should be parameters in the init, easier to make tests.